tests/test_TFUtil.py

```diff
@@ -469,6 +469,16 @@ def test_Data_copy_compatible_to_batch_feature_is_dynamic():
   assert_equal(t.get_size_dim_tag(0), dec.get_time_dim_tag())
 
 
+def test_Data_get_common_data_extra_static_spatial():
+  d1 = Data(name='t', shape=(None, 32, 128), dtype='float32', batch_dim_axis=0, time_dim_axis=1, feature_dim_axis=3,
+            auto_create_placeholders=True)
+  d2 = Data(name='r', shape=(None, 32, 128), dtype='float32', batch_dim_axis=0, time_dim_axis=1, feature_dim_axis=3,
+            auto_create_placeholders=True)
+  d2.get_size_dim_tag(0).declare_same_as(d1.get_size_dim_tag(0))
+  common = Data.get_common_data([d1, d2], warnings_out=sys.stdout)
+  assert d1.shape == common.shape
+
+
```

Comment:
Can you also add another further test, called `test_Data_get_common_data_extra2_static_spatial` or sth like that, where `d1/d2` has `shape=(None, 32, 32, 128)`? I'm not exactly sure if this behaves strange now with your change.

Reply:
(Also, btw, you set `batch_dim_axis=0, time_dim_axis=1, feature_dim_axis=3` explicitly, but I think you don't need that, or do you? Maybe remove that, to keep the test simple.)

Reply:
Done, that works fine

Reply:
> (Also, btw, you set `batch_dim_axis=0, time_dim_axis=1, feature_dim_axis=3` explicitly, but I think you don't need that, or do you? Maybe remove that, to keep the test simple.)

Done